### DISCUSSION TITLE
Docker Volumes - Docker activating

### DISCUSSION THREAD

**Discussion by ng1991** [Security-Onion-Solutions/securityonion#11072] [General]
Hi All,

This problem relates to https://github.com/Security-Onion-Solutions/securityonion/issues/9941 which is set to be resolved in 2.4.x but just wanted to highlight an issue it caused for us.

 We are running version 2.3.240 and I recently noticed on reboot the docker service takes around 10-20 mins to start. Looking at iotop " /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock" was reading the disk for a prolonged period of time and seemed to be the cause.

On investigation we identified the their was over 100 thousand volumes in /var/lib/docker/volumes. (docker volume ls | wc -l)

Once docker started we ran "docker volume prune" and this deleted all volumes and left 13 which is similar to the amount of containers (14) on a forward node.

After this completed I performed a reboot and docker started instantly and no longer had the 20 min delay on boot for SO containers to come up.

Just thought I would share this incase anyone is seeing similar behavior prior to the fix in 2.4.X, currently I am just going to create a cron job to clear any unused volumes.

**Comment by petiepooo** [comment]
I see #9941 has both 2.3 and 2.4 tags.

Devs, please make this a priority for 2.3 before it reaches EOL.
